### ISSUE TITLE
Implement cTracing interface

### ISSUE THREAD

**Issue by AndreasGocht** [score-p/scorep_binding_python#5]
Python provides a C tracing interface.

Implementation might speed-up the code a lot:

References:

## python3:
https://docs.python.org/3/c-api/init.html?highlight=pyeval_settrace#profiling-and-tracing
https://github.com/python/cpython/blob/3.6/Modules/_lsprof.c
https://github.com/python/cpython/blob/3.6/Include/frameobject.h

## python2:
https://docs.python.org/2.7/c-api/init.html?highlight=pyeval_settrace#profiling-and-tracing
https://github.com/python/cpython/blob/2.7/Modules/_lsprof.c
https://github.com/python/cpython/blob/2.7/Include/frameobject.h

**Comment by AndreasGocht** [comment]
Python implements the `cProfile` using _lsprof:

https://github.com/python/cpython/blob/master/Modules/_lsprof.c

```
static PyObject*
profiler_enable(ProfilerObject *self, PyObject *args, PyObject *kwds)
{
    int subcalls = -1;
    int builtins = -1;
    static char *kwlist[] = {"subcalls", "builtins", 0};
    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|ii:enable",
                                     kwlist, &subcalls, &builtins))
        return NULL;
    if (setSubcalls(self, subcalls) < 0 || setBuiltins(self, builtins) < 0) {
        return NULL;
    }

    PyThreadState *tstate = PyThreadState_GET();
    if (_PyEval_SetProfile(tstate, profiler_callback, (PyObject*)self) < 0) {
        return NULL;
    }

    self->flags |= POF_ENABLED;
    Py_RETURN_NONE;
}
```

They use `_PyEval_SetProfile`. However, I am not sure, that this is the way to go.

**Comment by Flamefire** [comment]
Fixed with #92
